chargeio/services/Merchant/MerchantService.cs

```diff
@@ -1,46 +1,37 @@
-﻿using ChargeIO.Infrastructure;
+﻿using ChargeIo.services;
+using ChargeIO.Infrastructure;
 using ChargeIO.Models;
 
 namespace ChargeIO.Services.Merchant
 {
-	public class MerchantService
-	{
-        private string SecretKey { get; }
-        
-        public MerchantService(string secretKey = "")
+    public sealed class MerchantService : ServiceBase
+    {
+        public MerchantService(string secretKey = null) : base(secretKey)
         {
-            if (secretKey == null)
-            {
-                SecretKey = Configuration.SecretKey;
-            }
-            else
-            {
-                SecretKey = secretKey.Length > 0 ? secretKey : Configuration.SecretKey;
-            }
         }
 
-        public virtual Models.Merchant GetMerchant()
-		{
-			var response = Requestor.GetString(Urls.Merchant, SecretKey);
+        public Models.Merchant GetMerchant()

```

Comment:
Why was the "virtual" keyword removed from all of the service methods (in all four service classes)? These were included to allow a consumer to potentially augment the behavior to suit their needs by subclassing our services.